### ISSUE TITLE
HAProxy 2.9.8 Crashing with FATAL: pool inconsistency detected in thread 1: tag mismatch on free().

### ISSUE THREAD

**Issue by derrickCSI** [haproxy/haproxy#2626]
### Detailed Description of the Problem

Since updating to version 2.9.8, haproxy has been crashing daily with the same stack trace and errors.

All crashes have the following message:

### Tag does not match (0x501b4b). Tag does not match any other pool.

root@dfw-lbl2_10.12.40.155 /var/log # grep Tag\ does messages*
messages:Jul  1 10:16:21 dfw-lbl2 haproxy[503860]: Tag does not match (0x501b4b). Tag does not match any other pool.
messages-20240630:Jun 26 10:50:36 XXXX haproxy[209633]: Tag does not match (0x501b4b). Tag does not match any other pool.
messages-20240630:Jun 27 11:06:37 XXXX haproxy[231847]: Tag does not match (0x501b4b). Tag does not match any other pool.
messages-20240630:Jun 28 13:42:55 XXXX haproxy[322739]: Tag does not match (0x501b4b). Tag does not match any other pool.
messages-20240630:Jun 29 11:49:55 XXXX haproxy[421634]: Tag does not match (0x501b4b). Tag does not match any other pool.

Latest stack trace from messages:

Jul  1 10:16:21 XXXXX haproxy[503860]: FATAL: pool inconsistency detected in thread 1: tag mismatch on free().
Jul  1 10:16:21 XXXXX haproxy[503860]:  caller: 0x501e56 (__stksess_kill+0xb6/0x135)
Jul  1 10:16:21 XXXXX haproxy[503860]:  item: 0x2a185e0
Jul  1 10:16:21 XXXXX haproxy[503860]:  pool: 0x23af280 ('h2s', size 168, real 176, users 2)
Jul  1 10:16:21 XXXXX haproxy[503860]: Tag does not match (0x501b4b). Tag does not match any other pool.
Jul  1 10:16:21 XXXXX haproxy[503860]: Contents around address 0x2a185e0+168=0x2a18688:
Jul  1 10:16:21 XXXXX haproxy[503860]:       0x2a18668 [00 00 00 00 00 00 00 00] [........]
Jul  1 10:16:21 XXXXX haproxy[503860]:       0x2a18670 [00 00 00 00 00 00 00 00] [........]
Jul  1 10:16:21 XXXXX haproxy[503860]:       0x2a18678 [08 00 9f 02 00 00 00 00] [........]
Jul  1 10:16:21 XXXXX haproxy[503860]:       0x2a18680 [e7 2b 79 cb a2 53 d9 f1] [.+y..S..]
Jul  1 10:16:21 XXXXX haproxy[503860]:       0x2a18688 [4b 1b 50 00 00 00 00 00] [K.P.....] [process_table_expire+0x28b/0x351]
Jul  1 10:16:21 XXXXX haproxy[503860]:       0x2a18690 [c0 00 00 00 00 00 00 00] [........]
Jul  1 10:16:21 XXXXX haproxy[503860]:       0x2a18698 [91 00 00 00 00 00 00 00] [........]
Jul  1 10:16:21 XXXXX haproxy[503860]:       0x2a186a0 [00 00 00 00 01 00 00 00] [........]
Jul  1 10:16:21 XXXXX haproxy[503860]:  call trace(14):
Jul  1 10:16:21 XXXXX haproxy[503860]:  |       0x5cead8 [0f 0b 66 0f 1f 44 00 00]: __pool_free+0x68/0x6a
Jul  1 10:16:21 XXXXX haproxy[503860]:  |       0x501e56 [85 ed 75 16 5b b8 01 00]: __stksess_kill+0xb6/0x135
Jul  1 10:16:21 XXXXX haproxy[503860]:  |       0x4d2a0f [4c 8b 04 24 0f 1f 44 00]: stream_free+0x75f/0x858
Jul  1 10:16:21 XXXXX haproxy[503860]:  |       0x4d7ee4 [48 83 7c 24 58 00 0f 84]: process_stream+0x3204/0x35a2
Jul  1 10:16:21 XXXXX haproxy[503860]:  |       0x5b731b [49 89 c7 4d 85 ff 74 77]: run_tasks_from_lists+0x33b/0x8e4
Jul  1 10:16:21 XXXXX haproxy[503860]:  |       0x5b7c30 [29 44 24 10 8b 4c 24 10]: process_runnable_tasks+0x360/0x6a3
Jul  1 10:16:21 XXXXX haproxy[503860]:  |       0x5827d8 [83 3d 8d 70 25 00 01 0f]: run_poll_loop+0x138/0x548
Jul  1 10:16:21 XXXXX haproxy[503860]:  |       0x582e07 [48 8b 1d f2 6e 24 00 48]: main+0x156897
Jul  1 10:16:21 XXXXX haproxy[503860]:  |       0x42dfa1 [31 c0 e8 c8 12 1d 00 e9]: main+0x1a31/0x2c04
Jul  1 10:16:21 XXXXX kernel: traps: haproxy[503860] trap invalid opcode ip:5cead8 sp:7ffdc5d38ae0 error:0 in haproxy[42a000+282000]

### Expected Behavior

Don't Crash.

### Steps to Reproduce the Behavior

Unknown steps to reproduce at this time. Will update if I am able to figure something out.

### Do you have any idea what may have caused this?

Upgraded to 2.9.8 from 2.5.0.

### What is your configuration?

```haproxy
global
    user haproxy
    group haproxy
    log 127.0.0.1   local0
    log 127.0.0.1   local1 notice
    maxconn 32000
    tune.ssl.default-dh-param 2048
    ssl-default-bind-ciphers AES128-GCM-SHA256:AES128-SHA256:AES256-GCM-SHA384:AES256-SHA256:CAMELLIA:DHE-DSS-AES128-GCM-SHA256:DHE-DSS-AES128-SHA256:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES128-SHA256:DHE-RSA-AES256-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-SHA384:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-RSA-AES256-SHA:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:DHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA:DHE-DSS-AES256-SHA:AES256-SHA:AES128-SHA

    ssl-default-bind-ciphersuites TLS_AES_128_GCM_SHA256:TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256
    ssl-default-bind-options prefer-client-ciphers no-sslv3 no-tlsv10 no-tlsv11 no-tls-tickets

    ssl-default-server-ciphers AES128-GCM-SHA256:AES128-SHA256:AES256-GCM-SHA384:AES256-SHA256:CAMELLIA:DHE-DSS-AES128-GCM-SHA256:DHE-DSS-AES128-SHA256:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES128-SHA256:DHE-RSA-AES256-SHA256:ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES128-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-ECDSA-AES256-SHA384:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-SHA256:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-SHA384:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-CHACHA20-POLY1305:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:ECDHE-RSA-AES256-SHA:ECDHE-RSA-AES128-SHA:ECDHE-ECDSA-AES128-SHA:DHE-RSA-AES256-SHA:DHE-RSA-AES128-SHA:DHE-DSS-AES256-SHA:AES256-SHA:AES128-SHA

    ssl-default-server-ciphersuites TLS_AES_128_GCM_SHA256:TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256
    ssl-default-server-options no-sslv3 no-tlsv10 no-tlsv11 no-tls-tickets

    h1-case-adjust content-disposition Content-Disposition
    stats socket /var/run/haproxy.stat mode 666 level admin

defaults
    log     global
    mode    http
    option  forwardfor
    option  httplog
    option  http-server-close
    timeout http-keep-alive 5s
    timeout http-request 60000ms
    timeout client 60000ms
    timeout server 240000ms
    timeout connect 5000ms
    option h1-case-adjust-bogus-client

    # handle the situation where a client suddenly disappears from the net
    timeout client-fin 30s

    # sorry page to return when backend returns 503 - when all app nodes are disabled for maint.
    errorfile 503 /etc/haproxy/errors/503sorry.http

listen stats
    bind XX.XX.XX.XX:2007
    mode            http
    log             global
    maxconn 10
    timeout connect 100s
    timeout client  100s
    timeout server  100s
    timeout queue   100s
    stats enable
    stats hide-version
    stats refresh 30s
    stats show-node
    stats auth ##############
    stats uri  /stats
```

### Output of `haproxy -vv`

```plain
# haproxy -vv
HAProxy version 2.9.8-a140abd 2024/06/11 - https://haproxy.org/
Status: stable branch - will stop receiving fixes around Q1 2025.
Known bugs: http://www.haproxy.org/bugs/bugs-2.9.8.html
Running on: Linux 5.14.0-427.22.1.el9_4.x86_64 #1 SMP PREEMPT_DYNAMIC Wed Jun 19 17:35:04 UTC 2024 x86_64
Build options :
  TARGET  = linux-glibc
  CPU     = generic
  CC      = cc
  CFLAGS  = -O2 -g -Wall -Wextra -Wundef -Wdeclaration-after-statement -Wfatal-errors -Wtype-limits -Wshift-negative-value -Wshift-overflow=2 -Wduplicated-cond -Wnull-dereference -fwrapv -Wno-address-of-packed-member -Wno-unused-label -Wno-sign-compare -Wno-unused-parameter -Wno-clobbered -Wno-missing-field-initializers -Wno-cast-function-type -Wno-string-plus-int -Wno-atomic-alignment
  OPTIONS = USE_OPENSSL=1 USE_LUA=1 USE_ZLIB=1 USE_SYSTEMD=1 USE_STATIC_PCRE2=1
  DEBUG   = -DDEBUG_STRICT -DDEBUG_MEMORY_POOLS

Feature list : -51DEGREES +ACCEPT4 +BACKTRACE -CLOSEFROM +CPU_AFFINITY +CRYPT_H -DEVICEATLAS +DL -ENGINE +EPOLL -EVPORTS +GETADDRINFO -KQUEUE -LIBATOMIC +LIBCRYPT +LINUX_CAP +LINUX_SPLICE +LINUX_TPROXY +LUA +MATH -MEMORY_PROFILING +NETFILTER +NS -OBSOLETE_LINKER +OPENSSL -OPENSSL_AWSLC -OPENSSL_WOLFSSL -OT -PCRE +PCRE2 -PCRE2_JIT -PCRE_JIT +POLL +PRCTL -PROCCTL -PROMEX -PTHREAD_EMULATION -QUIC -QUIC_OPENSSL_COMPAT +RT +SHM_OPEN -SLZ +SSL -STATIC_PCRE +STATIC_PCRE2 +SYSTEMD +TFO +THREAD +THREAD_DUMP +TPROXY -WURFL +ZLIB

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_TGROUPS=16, MAX_THREADS=256, default=2).
Built with OpenSSL version : OpenSSL 3.0.14 4 Jun 2024
Running on OpenSSL version : OpenSSL 3.0.7 1 Nov 2022
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : TLSv1.0 TLSv1.1 TLSv1.2 TLSv1.3
OpenSSL providers loaded : default
Built with Lua version : Lua 5.4.6
Built with network namespace support.
Built with zlib version : 1.2.11
Running on zlib version : 1.2.11
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built with PCRE2 version : 10.40 2022-04-14
PCRE2 library supports JIT : no (USE_PCRE2_JIT not set)
Encrypted password support via crypt(3): yes
Built with gcc compiler version 11.4.1 20231218 (Red Hat 11.4.1-3)

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
         h2 : mode=HTTP  side=FE|BE  mux=H2    flags=HTX|HOL_RISK|NO_UPG
       fcgi : mode=HTTP  side=BE     mux=FCGI  flags=HTX|HOL_RISK|NO_UPG
  <default> : mode=HTTP  side=FE|BE  mux=H1    flags=HTX
         h1 : mode=HTTP  side=FE|BE  mux=H1    flags=HTX|NO_UPG
  <default> : mode=TCP   side=FE|BE  mux=PASS  flags=
       none : mode=TCP   side=FE|BE  mux=PASS  flags=NO_UPG

Available services : none

Available filters :
        [BWLIM] bwlim-in
        [BWLIM] bwlim-out
        [CACHE] cache
        [COMP] compression
        [FCGI] fcgi-app
        [SPOE] spoe
        [TRACE] trace
```

### Last Outputs and Backtraces

```plain
No coredump available.  System has been reconfigured to capture if it happens again.
```

**Comment by capflam** [comment]
It is most probably the same than #2611. I will backport the fix tomorrow to 2.9.

**Comment by capflam** [comment]
Now backported to 2.9. I'm closing for now because it really seems to be a duplicate of #2611. If I'm wrong and it still crash with the fix, please, reopen the issue. Thanks !
